sdk/xmake.lua

```diff
@@ -683,6 +683,36 @@ rule("firmware")
 			end
 		end)
 
+		local shared_objects = {}
+		visit_all_dependencies(function (target)
+			local globals = target:values("shared_objects")
+			if globals then
+				for name, size in pairs(globals) do
+					if not (name == "__wrap_locked__") then
+						if shared_objects[global] and (not (shared_objects[global] == size)) then
+							raise("Global " .. global .. " is declared with different sizes.")
+						end
+						shared_objects[name] = size
+					end
+				end
+			end
+		end)
+		-- TODO: We should sort pre-shared globals by size to minimise padding.
+		-- Each global is emitted as a separate section so that we can use
+		-- CAPALIGN and let the linker insert the required padding.
+		local shared_objects_template =
+			"\n\t\t__cheriot_shared_object_section_${global} : CAPALIGN" ..
+			"\n\t\t{" ..
+			"\n\t\t\t__cheriot_shared_object_${global} = .;" ..
+			"\n\t\t\t. += ${size};" ..
+			"\n\t\t\t__cheriot_shared_object_${global}_end = .;" ..
+			"\n\t\t}\n"
```

Comment:
Could we generalize the template to permit `__export_mem_hazard_pointers` and `__export_mem_allocator_epoch` to also be included in the `shared_objects` table, so that they get sorted for size along with the user defined ones?  Or, could we make those objects use this generic mechanism instead?

Reply:
Yes, next step is to make these use the new mechanism.

Reply:
Added in the next commit.